File: win-prog/ProgWinCS/Tree View and List View/SplitThreeAcross/SplitThreeAcross.cs
//-----------------------------------------------
// SplitThreeAcross.cs © 2001 by Charles Petzold
//-----------------------------------------------
using System;
using System.Drawing;
using System.Windows.Forms;

class SplitThreeAcross: Form
{
     public static void Main()
     {
          Application.Run(new SplitThreeAcross());
     }
     public SplitThreeAcross()
     {
          Text = "Split Three Across";

          Panel panel1     = new Panel();
          panel1.Parent    = this;
          panel1.Dock      = DockStyle.Fill;
          panel1.BackColor = Color.Cyan;
          panel1.Resize   += new EventHandler(PanelOnResize);
          panel1.Paint    += new PaintEventHandler(PanelOnPaint);

          Splitter split1  = new Splitter();
          split1.Parent    = this;
          split1.Dock      = DockStyle.Left;

          Panel panel2     = new Panel();
          panel2.Parent    = this;
          panel2.Dock      = DockStyle.Left;
          panel2.BackColor = Color.Lime;
          panel2.Resize   += new EventHandler(PanelOnResize);
          panel2.Paint    += new PaintEventHandler(PanelOnPaint);

          Splitter split2  = new Splitter();
          split2.Parent    = this;
          split2.Dock      = DockStyle.Right;

          Panel panel3     = new Panel();
          panel3.Parent    = this;
          panel3.Dock      = DockStyle.Right;
          panel3.BackColor = Color.Red;
          panel3.Resize   += new EventHandler(PanelOnResize);
          panel3.Paint    += new PaintEventHandler(PanelOnPaint);

          panel1.Width = 
          panel2.Width = 
          panel3.Width = ClientSize.Width / 3;
     }
     void PanelOnResize(object obj, EventArgs ea)
     {
          ((Panel) obj).Invalidate();
     }
     void PanelOnPaint(object obj, PaintEventArgs pea)
     {
          Panel    panel = (Panel) obj;
          Graphics grfx  = pea.Graphics;

          grfx.DrawEllipse(Pens.Black, 0, 0, 
                           panel.Width - 1, panel.Height - 1);
     }
}

File: win-prog/ProgWinVB/Tree View and List View/SplitThreeAcross/SplitThreeAcross.vb
'-------------------------------------------------
' SplitThreeAcross.vb (c) 2002 by Charles Petzold
'-------------------------------------------------
Imports System
Imports System.Drawing
Imports System.Windows.Forms

Class SplitThreeAcross
    Inherits Form

    Shared Sub Main()
        Application.Run(new SplitThreeAcross())
    End Sub

    Sub New()
        Text = "Split Three Across"

        Dim pnl1 As New Panel()
        pnl1.Parent = Me
        pnl1.Dock = DockStyle.Fill
        pnl1.BackColor = Color.Cyan
        AddHandler pnl1.Resize, AddressOf PanelOnResize
        AddHandler pnl1.Paint, AddressOf PanelOnPaint

        Dim split1 As New Splitter()
        split1.Parent = Me
        split1.Dock = DockStyle.Left

        Dim pnl2 As New Panel()
        pnl2.Parent = Me
        pnl2.Dock = DockStyle.Left
        pnl2.BackColor = Color.Lime
        AddHandler pnl2.Resize, AddressOf PanelOnResize
        AddHandler pnl2.Paint, AddressOf PanelOnPaint

        Dim split2 As New Splitter()
        split2.Parent = Me
        split2.Dock = DockStyle.Right

        Dim pnl3 As New Panel()
        pnl3.Parent = Me
        pnl3.Dock = DockStyle.Right
        pnl3.BackColor = Color.Red
        AddHandler pnl3.Resize, AddressOf PanelOnResize
        AddHandler pnl3.Paint, AddressOf PanelOnPaint

        pnl1.Width = ClientSize.Width \ 3
        pnl2.Width = ClientSize.Width \ 3
        pnl3.Width = ClientSize.Width \ 3
    End Sub

    Private Sub PanelOnResize(ByVal obj As Object, ByVal ea As EventArgs)
        DirectCast(obj, Panel).Invalidate()
    End Sub

    Private Sub PanelOnPaint(ByVal obj As Object, _
                             ByVal pea As PaintEventArgs)
        Dim pnl As Panel = DirectCast(obj, Panel)
        Dim grfx As Graphics = pea.Graphics

        grfx.DrawEllipse(Pens.Black, 0, 0, _
                         pnl.Width - 1, pnl.Height - 1)
    End Sub
End Class
